spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/SparkTransformConverter.java

```diff
@@ -0,0 +1,236 @@
+
+/**
+ * SparkTransformConverter translate between Spark transform and Gravitino partition, distribution,
+ * sort orders. There may be multi partition transforms, but should be only one bucket transform.
+ * Spark bucket transform is corresponding to Gravitino Hash distribution without sort orders. Spark
+ * sorted bucket transform is corresponding to Gravitino Hash distribution with sort orders.
+ */
+public class SparkTransformConverter {
+
+  public static class DistributionAndSortOrdersInfo {
+    @Getter private Distribution distribution;
+    @Getter private SortOrder[] sortOrders;
+
+    private void setDistribution(Distribution distributionInfo) {
+      Preconditions.checkState(distribution == null, "Should only set distribution once");
+      this.distribution = distributionInfo;
+    }
+
+    private void setSortOrders(SortOrder[] sortOrdersInfo) {
+      Preconditions.checkState(sortOrders == null, "Should only set sort orders once");
+      this.sortOrders = sortOrdersInfo;
+    }
+  }
+
+  public static Transform[] toGravitinoPartitions(
+      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+    if (ArrayUtils.isEmpty(transforms)) {
+      return new Transform[0];
+    }
+
+    return Arrays.stream(transforms)
+        .filter(
+            transform ->
+                !((transform instanceof BucketTransform)
+                    || (transform instanceof SortedBucketTransform)))
+        .map(
+            transform -> {
+              if (transform instanceof IdentityTransform) {
+                IdentityTransform identityTransform = (IdentityTransform) transform;
+                return Transforms.identity(identityTransform.reference().fieldNames());
+              } else {
+                throw new NotSupportedException(
+                    "Doesn't support Spark transform: " + transform.name());
+              }
+            })
+        .toArray(Transform[]::new);
+  }
+
+  public static DistributionAndSortOrdersInfo toGravitinoDistributionAndSortOrders(
+      org.apache.spark.sql.connector.expressions.Transform[] transforms) {
+    DistributionAndSortOrdersInfo bundles = new DistributionAndSortOrdersInfo();
+    if (ArrayUtils.isEmpty(transforms)) {
+      return bundles;
+    }
+
+    Arrays.stream(transforms)
+        .forEach(
+            transform -> {
+              if (transform instanceof SortedBucketTransform) {
+                Pair<Distribution, SortOrder[]> pair =
+                    toGravitinoDistributionAndSortOrders((SortedBucketTransform) transform);
+                bundles.setDistribution(pair.getLeft());
+                bundles.setSortOrders(pair.getRight());
+              } else if (transform instanceof BucketTransform) {
+                BucketTransform bucketTransform = (BucketTransform) transform;
+                Distribution distribution = toGravitinoDistribution(bucketTransform);
+                bundles.setDistribution(distribution);
+              }
```

Comment:
no, spark just supports two kinds of bucketTransform,  the else part maybe partition transform

Reply:
If Spark supports the third type without else here, what will the user receive?

Reply:
`toGravitinoPartitions` will fail,  we could handle this easily if using formal `toGravitinoTransform` way, after separating two interfaces do you have any good ideas to handle this?

Reply:
I think adding `else` statement to throw an unsupported exception is enough here

Reply:
it couldn't be handled like this,  because the transforms contains all partition and bucket transforms. Another solution may split transforms to partition and bucket first, but this makes the code complicated, is it nessesery?

Reply:
add filter and else logic, @mchades please review again

Reply:
OK, this part is clearer now